node/pkg/guardiansigner/benchmarksigner.go

```diff
@@ -0,0 +1,78 @@
+package guardiansigner
+
+/*
+	The Benchmark signer is a type of signer that wraps other signers,
+	recording the latency of signing and signature verification into
+	histograms. As additional signers are implemented, relying on 3rd
+	party services, benchmarking signers is useful to ensure observation
+	signing happens at an acceptable rate.
+*/
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+// The BenchmarkSigner is a signer that wraps other signers, recording the latency of
+// signing and signature verification through prometheus histograms.
+type BenchmarkSigner struct {
+	innerSigner GuardianSigner
+}
+
+var (
+	guardianSignerSigningLatency = promauto.NewHistogram(
+		prometheus.HistogramOpts{
+			Name:    "wormhole_guardian_signer_signing_latency_us",
+			Help:    "Latency histogram for Guardian signing requests",
+			Buckets: []float64{10.0, 20.0, 50.0, 100.0, 1000.0, 5000.0, 10000.0, 100_000.0, 1_000_000.0, 10_000_000.0, 100_000_000.0, 1_000_000_000.0},
+		})
+
+	guardianSignerVerifyLatency = promauto.NewHistogram(
+		prometheus.HistogramOpts{
+			Name:    "wormhole_guardian_signer_sig_verify_latency_us",
+			Help:    "Latency histogram for Guardian signature verification requests",
+			Buckets: []float64{10.0, 20.0, 50.0, 100.0, 1000.0, 5000.0, 10000.0, 100_000.0, 1_000_000.0, 10_000_000.0, 100_000_000.0, 1_000_000_000.0},
+		})
+)
+
+func BenchmarkWrappedSigner(innerSigner GuardianSigner) *BenchmarkSigner {
+	if innerSigner == nil {
+		return nil
+	}
+
+	return &BenchmarkSigner{
+		innerSigner: innerSigner,
+	}
+}
+
+func (b *BenchmarkSigner) Sign(ctx context.Context, hash []byte) ([]byte, error) {
```

Comment:
Will errors skew the data? Should you add a metric for errors and bucket the latency separately or exclude them from the latency numbers?

Reply:
Thanks for this! I've added error counters.

Reply:
Following on from Bruce's comment, you're now tracking error counts but the latency figures aren't grouped by success/fail cases. Should we only observe the latency for a successful operation?

Reply:
>Should we only observe the latency for a successful operation?

We definitely should.

Reply:
Changed this. Thanks for the suggestion.